numpy/lib/tests/test_arraypad.py

```diff
@@ -344,6 +347,19 @@ def test_check_mean_2(self):
             )
         assert_array_equal(a, b)
 
+    @pytest.mark.parametrize("mode", [
+        pytest.mark.xfail(reason="gh-11216")("mean"),
+        "median",
+        "minimum",
+        "maximum"
+    ])
+    def test_same_prepend_append(self, mode):
+        # Check if the prepended and appended values are the same.
+        # Regression test for issue gh-11216
+        a = np.array([-1, 2, -1]) + np.array([0, 1e-12, 0], dtype=np.float64)
```

Comment:
I think it is best if we keep this test just for 'mean'. The other functions will not suffer from this bug.

Reply:
Why not test them anyway?

Reply:
I get it, but this is very convoluted targeted test. 
I think it can definitely make use of a "in source" comment summarizing the issue.

We are essentially testing the implementation of the algorithm as well as the peculiarities of floating point numbers. it is easy to understand why `maximum` and `minimum` are not likely to suffer from this bug. `median` will also not suffer from this.

If we want to test the general behaviour for all 4 other modes, I think it is better to have them in a more general different test and keep this particular test very targeted.